### ISSUE TITLE
LowercaseUrls = true in RouteOptions doesn't affect URLs with non-Roman characters

### ISSUE THREAD

**Issue by DevelAx** [aspnet/Routing#435]
In ASP.NET MVC CORE 2.0

```
public void ConfigureServices(IServiceCollection services)
{
    services.Configure<RouteOptions>(options => options.LowercaseUrls = true);
    services.AddMvc();
}
```

`<a asp-action="New-Post">New post</a>`

works fine giving URL like http://domain/controller/new-post
But this one

`<a asp-action="Новая-Публикация">Новая публикация</a>`

produces URL like http://domain/controller/Новая-Публикация

**Seems like RouteCollection uses ToLowerInvariant() to lower-case the URL:** https://github.com/aspnet/Routing/blob/032bcf43b2cefe641fc6ee9ef3ab0769024a182c/src/Microsoft.AspNetCore.Routing/RouteCollection.cs#L155

**Comment by Eilon** [comment]
Ah, I think the problem here is that the Cyrillic part of the URL is already URL-encoded (to `%xx%xx%xx` etc.), so the `ToLowerInvariant()` doesn't have much to do at that point.

I think to fix this we'd need to have somewhat different logic about how/where to do the lowercase operation.

**Comment by develax** [comment]
Thank you for the clarification, Eilon! 
Should I expect it will be fixed in the future?
In modern browsers encoded URLs are displayed in a friendly decoded form.

**Comment by Eilon** [comment]
@DevelAx we'll investigate this issue some more to see what we can do. I'm not sure how easy it will be. Yes, browsers typically show the "friendly decoded" URL, which is why I had to look at the "view source" of the page to get a hint of what was going on.

A not-so-fun workaround in the meantime is to just use lowercase letters everywhere 😄

**Comment by DevelAx** [comment]
Ok, thank you!
Sure you will find the most graceful way to fix it.

**Comment by Eilon** [comment]
Moving to Backlog for now because this turns out to be quite complicated to fix.

However, we're looking at some bigger changes to Routing in the near future and we'll try to account for fixing this in the new design.

**Comment by aspnet-hello** [comment]
This issue was moved to aspnet/Home#2665

**Comment by rynowak** [comment]
I'm reopening this for consideration as part of the current routing work. Options that we provide today to process the generated URLs don't always behave predictably

**Comment by mkArtakMSFT** [comment]
@rynowak, is this something you plan to get fixed as part of 2.2 work?

**Comment by rynowak** [comment]
This has been fixed for the new Endpoint Routing feature set in 2.2. Old `IRouter` still behaves as is.
